### PR TITLE
[CIAPP-3594] Explain custom tags for CI visibility circleci integration

### PR DESCRIPTION
**Pull request by AdrianLC**
Extracted from https://github.com/DataDog/documentation/pull/12994

### What does this PR do?

Explain how to provide custom tags to CI Visibility for pipelines. This is supported through a new query parameter in the webhooks.

This PR documents how to use the new parameter with webhooks while configuring the webhook in CircleCI.

### Preview

- https://docs-staging.datadoghq.com/adrian.lopezcalvo%2Fci-visibility-circleci-custom-tags/continuous_integration/setup_pipelines/circleci/

### Additional Notes

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/continuous_integration/setup_pipelines/circleci.md [hunk 1]
```diff
@@ -29,6 +29,9 @@ The Datadog integration for [CircleCI][1] works by using [webhooks][2] to send d
 
 2. Click **Add Webhook** to save the new webhook.
```
Comment (fermayo) [inline content/en/continuous_integration/setup_pipelines/circleci.md:31]: Maybe add a subsection here like this?

```suggestion

### Setting custom tags
```

File: content/en/continuous_integration/setup_pipelines/circleci.md [hunk 2]
```diff
@@ -29,6 +29,9 @@ The Datadog integration for [CircleCI][1] works by using [webhooks][2] to send d
...
 
+To set custom tags, add to the **Webhook URL** a URL-encoded query parameter `tags`, with `key:value` pairs separated by commas. If the tag pair contains any commas, quotes are needed. For example, to add `key1:value1,"key2: value with , comma",key3:value3`, the following string would need to be appended to the
+**Webhook URL**: `?tags=key1%3Avalue1%2C%22key2%3A+value+with+%2C+comma%22%2Ckey3%3Avalue3`
```
Comment (fermayo) [inline content/en/continuous_integration/setup_pipelines/circleci.md:33]:
```suggestion
To set custom tags to all the pipeline and job spans generated by the integration, add to the **Webhook URL** a URL-encoded query parameter `tags`, with `key:value` pairs separated by commas. If the tag pair contains any commas, quotes are needed. For example, to add `key1:value1,"key2: value with , comma",key3:value3`, the following string would need to be appended to the **Webhook URL**: `?tags=key1%3Avalue1%2C%22key2%3A+value+with+%2C+comma%22%2Ckey3%3Avalue3`
```

**Review by fermayo — APPROVED** [review]

File: content/en/continuous_integration/setup_pipelines/circleci.md (hunk 2 above)
Comment (AdrianLC) [reply content/en/continuous_integration/setup_pipelines/circleci.md:34]: did you just put everything in the same line? I don't think that changes anything in the rendered markdown

**Review by urseberry — COMMENTED** [review]
Screen Shot 2022-02-09 at 2 00 07 PM

Here is the correct link to the preview. https://docs-staging.datadoghq.com/adrian.lopezcalvo/ci-visibility-circleci-custom-tags/continuous_integration/setup_pipelines/circleci/#setting-custom-tags Is the text showing up the way you want?

**Comment by fermayo** [comment]
> > Screen Shot 2022-02-09 at 2 00 07 PM
> > Here is the correct link to the preview. https://docs-staging.datadoghq.com/adrian.lopezcalvo/ci-visibility-circleci-custom-tags/continuous_integration/setup_pipelines/circleci/#setting-custom-tags Is the text showing up the way you want?
> 
> yes, that looks good

Maybe put the string in a separate section so it does not break between two lines?

**Comment by AdrianLC** [comment]
> > > Screen Shot 2022-02-09 at 2 00 07 PM
> > > Here is the correct link to the preview. https://docs-staging.datadoghq.com/adrian.lopezcalvo/ci-visibility-circleci-custom-tags/continuous_integration/setup_pipelines/circleci/#setting-custom-tags Is the text showing up the way you want?
> > 
> > 
> > yes, that looks good
> 
> Maybe put the string in a separate section so it does not break between two lines?

Ah yes I hadn't notice. Done

File: content/en/continuous_integration/setup_pipelines/circleci.md [hunk 3]
```diff
@@ -29,6 +29,11 @@ The Datadog integration for [CircleCI][1] works by using [webhooks][2] to send d
 
 2. Click **Add Webhook** to save the new webhook.
 
+### Setting custom tags
+To set custom tags to all the pipeline and job spans generated by the integration, add to the **Webhook URL** a URL-encoded query parameter `tags`, with `key:value` pairs separated by commas. If the tag pair contains any commas, quotes are needed. For example, to add `key1:value1,"key2: value with , comma",key3:value3`, the following string would need to be appended to the **Webhook URL**:
```
Comment (urseberry) [inline content/en/continuous_integration/setup_pipelines/circleci.md:33]: I have one more wording suggestion to make this part clearer:

```suggestion
To set custom tags to all the pipeline and job spans generated by the integration, add to the **Webhook URL** a URL-encoded query parameter `tags`, with `key:value` pairs separated by commas. If a key:value pair contains any commas, surround it with quotes. For example, to add `key1:value1,"key2: value with , comma",key3:value3`, the following string would need to be appended to the **Webhook URL**:
```

**Review by urseberry — APPROVED** [review]

**Pull request merged**